packages/compass-import-export/src/components/import-toast.tsx

```diff
@@ -0,0 +1,255 @@
+    ),
+    variant: 'warning',
+  });
+}
+
+function showCancelledToast({
+  errorLogFilePath,
+  errors,
+}: {
+  errorLogFilePath: string;
+  errors: Error[];
+}) {
+  if (errors.length > 0) {
+    const statusMessage = getToastErrorsText(errors);
+    openToast(importToastId, {
+      title: 'Import aborted with the following errors:',
+      body: (
+        <ImportToastBody
+          statusMessage={statusMessage}
+          actionHandler={() => void openFile(errorLogFilePath)}
+          actionText="view log"
+        />
+      ),
+      variant: 'warning',
+    });
+    return;
+  }
+
+  openToast(importToastId, {
+    title: 'Import aborted.',
+    body: null,
+    variant: 'warning',
+  });
+}
+
+function showFailedToast(err: Error | undefined) {
+  openToast(importToastId, {
+    title: 'Failed to import with the following error:',
+    body: err?.message,
+    variant: 'warning',
+  });
+}
+
+function useImportToast({
+  cancelImport,
+  docsWritten,
+  errorLogFilePath,
+  docsProcessed,
+  status,
+  errors,
+  fileName,
+}: {
+  cancelImport: () => void;
+  docsWritten: number;
+  errorLogFilePath: string;
+  docsProcessed: number;
+  status: ProcessStatus;
+  errors: Error[];
+  fileName: string;
+}) {
+  useEffect(() => {
+    if (status === 'STARTED') {
+      if (docsProcessed > 0) {
+        showInProgressToast({
+          cancelImport,
+          docsWritten,
+          fileName,
+        });
+      } else {
+        showStartingToast({
+          cancelImport,
+          fileName,
+        });
+      }
+    } else if (status === 'COMPLETED') {
+      if (errors.length > 0) {
+        showCompletedWithErrorsToast({
+          docsWritten,
+          errors,
+          docsProcessed,
+          errorLogFilePath: errorLogFilePath,
+        });
+      } else {
+        showCompletedToast({
+          docsWritten,
+        });
+      }
+    } else if (status === 'CANCELED') {
+      showCancelledToast({
+        errors,
+        errorLogFilePath: errorLogFilePath,
+      });
+    } else if (status === 'FAILED') {
+      showFailedToast(errors[0]);
+    }
+  }, [
+    status,
+    docsProcessed,
+    docsWritten,
+    fileName,
+    errorLogFilePath,
+    errors,
+    cancelImport,
+  ]);
+
+  const abortIfInProgress = useRef(() => {
+    /* noop */
+  });
+  useEffect(() => {
+    abortIfInProgress.current = () => {
+      // When the component is dismounted, we abort the import if
+      // it's in progress and update the toast.
+      if (status === 'STARTED') {
+        cancelImport();
+        showCancelledToast({
+          errors,
+          errorLogFilePath: errorLogFilePath,
+        });
+      }
+    };
+  }, [errors, errorLogFilePath, cancelImport, status]);
+
+  useEffect(() => {
+    return () => {
+      // Abort the import operation when it's in progress and the import is going away.
+      abortIfInProgress.current?.();
+    };
+  }, []);
+
+  // We return null as we're using this hook as a standalone component.
```

Comment:
I am not sure I follow why is this a component in the first place? All the effects here seem to be just parts of the async action flow that actually does the import, so feel like this belongs in the store, not here

Reply:
Thanks for bringing this up. Moved the handling of these toast functions to the import flow like they were before. The rendering of the toasts is still in this `components/import-toast.tsx` file, however it's not directly called from the import async flow. Does that make sense to you? Is there a better place?

Reply:
I think you found the best way to split it, nice!